### ISSUE TITLE
Generic types in annotations of __getattr__ in metaclasses are not deduced correctly 

### ISSUE THREAD

**Issue by dedebenui** [microsoft/pylance-release#1481]
<h3> Environment data </h3>

-   Language Server version: 2021.6.2
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): Python 3.9.5 (Anaconda)
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: off

<h3> Expected behaviour </h3>

Pylance should determine the type of all 4 outputs to be `int`

<h3> Actual behaviour </h3>

Pylance determines that the type of `output3` is `Type[SomeFunctions]` no matter the type of the argument to `SomeFunctions.other_function`.

<h3> Code Snippet / Additional information </h3>

<img width="440" alt="Capture d’écran 2021-06-22 à 11 09 56" src="https://user-images.githubusercontent.com/41047847/122897554-675b2800-d34a-11eb-8e77-3d972dbcf4e5.png">
<img width="416" alt="Capture d’écran 2021-06-22 à 11 11 08" src="https://user-images.githubusercontent.com/41047847/122897724-8d80c800-d34a-11eb-8991-7d37366fba8f.png">

```python
from typing import Callable, TypeVar

T = TypeVar("T")

class FunctionCollection(type):
    def __getattr__(cls, key) -> Callable[[T], T]:
        return lambda x: x

    def function(cls, arg: T) -> T:
        return arg

    def factory(cls) -> Callable[[T], T]:
        return lambda x: x

class SomeFunctions(metaclass=FunctionCollection):
    pass


output1 = SomeFunctions.function(3)
output2 = SomeFunctions.factory()(3)
output3 = SomeFunctions.other_function(3)
output4 = SomeFunctions.__getattr__("other_function")(3)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

The problem is that the type evaluator assumed that the function returned by the metaclass `__getattr__` needed to be bound to the function `SomeFunctions`. It therefore assumed that the first parameter (of type `T`) was a `self` parameter. The Python interpreter appears to know that the return result from `__getattr__` should not be treated like other class or instance methods and therefore does not need to be bound by the caller's class, nor does it receive a `self` argument.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202170-6-july-2021
